sklearn/decomposition/tests/test_pca.py

```diff
@@ -256,34 +257,33 @@ def test_singular_values():
 
     X = rng.randn(n_samples, n_features)
 
-    pca = PCA(n_components=2, svd_solver='full',
-              random_state=rng).fit(X)
-    apca = PCA(n_components=2, svd_solver='arpack',
+    pca = PCA(n_components=2, svd_solver='full', random_state=rng).fit(X)
+    apca = PCA(n_components=2, svd_solver='arpack', random_state=rng).fit(X)
+    # Increase the number of power iterations to get greater accuracy in tests
+    rpca = PCA(n_components=2, svd_solver='randomized', iterated_power=40,
```

Comment:
This whole `test_singular_values()` test should not be changed IMO.

If you have to set `iterated_power=40` for the test to pass now, that's a regression, and that confirms my suggestion: `randomized_pca` should only be called for sparse matrices. Dense matrices should still use `randomized_svd`.

Also in general, please avoid style fixes that are not related to the issue (like spaces between ` ** `, etc). I know you mean well but then it makes it harder to determine what changes are important during the review.